scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
 	}
 	return ret;
 }
+
+// TODO: remove mutual dependencies and move the render engine into it's own cc and header files because it's getting hard to navigate the code base.
+// PDF never deviates from the line when it comes to colenear
+bool TextRegion::coLinera(qreal a, qreal b)
+{
+	return abs(a - b) < 1 ? true : false;
+}
+
+// like _colenia but we allow a deviation of upto +-2 rejion font text widths
+bool TextRegion::closeToX(qreal x1, qreal x2)
+{
+	//TODO: return abs(x1 - x2) <= coreText.mWidth() * 2 ? true : false;
+	return abs(x2 - x1) <= lineSpacing * 6 ? true : abs(x1 - this->textRegioBasenOrigin.x()) <= lineSpacing ? true: false; //allow infinate overrun but only one char width underrun

```

Comment:
try not to concatenate `? : `s.  
in this case an if with an early return is probably the better choice...

Reply:
if you want a one-liner, an alternative is probably (you should check):

```cpp
//allow infinite overrun but only one char width underrun
bool isCloseToX(...)
{
    return (abs(x2 - x1) <= lineSpacing * 6) || (abs(x1 - this->textRegioBasenOrigin.x()) <= lineSpacing);
}
```

still no idea what it does but it's a bit more readable

Reply:
it returns true if the two points are close to each other in the x dimention but in a fuzzy way.